src/rules/linalg/factorization.jl

```diff
@@ -79,3 +82,192 @@ function _add!(X::AbstractMatrix{T}, Y::AbstractMatrix{T}) where T<:Real
     end
     X
 end
+
+#####
+##### `cholesky`
+#####
+
+function rrule(::typeof(cholesky), X::AbstractMatrix{<:Real})
+    F = cholesky(X)
+    ∂X = Rule(Ȳ->chol_blocked_rev(Matrix(Ȳ), Matrix(F.U), 25, true))
+    return F, ∂X
+end
+
+function rrule(::typeof(getproperty), F::Cholesky, x::Symbol)
+    if x === :U
+        if F.uplo === 'U'
+            ∂F = Ȳ->UpperTriangular(Ȳ)
+        else
+            ∂F = Ȳ->LowerTriangular(Ȳ')
+        end
+    elseif x === :L
+        if F.uplo === 'L'
+            ∂F = Ȳ->LowerTriangular(Ȳ)
+        else
+            ∂F = Ȳ->UpperTriangular(Ȳ')
+        end
+    end
+    return getproperty(F, x), (∂F, DNERule())
+end
+
+# TODO: The comment below about row- versus column-major is incorrect; Julia's arrays are
+# column-major. This comment along with the below implementation was copied from Nabla and
+# works, but the implementation should perhaps should be revisited to ensure we're actually
+# being cache friendly.
+#
+# See [1] for implementation details: pages 5-9 in particular. The derivations presented in
+# [1] assume column-major layout, whereas Julia primarily uses row-major. We therefore
+# implement both the derivations in [1] and their transpose, which is more appropriate to
+# Julia.
+#
+# [1] - "Differentiation of the Cholesky decomposition", Murray 2016
+
+"""
+    level2partition(A::AbstractMatrix, j::Integer, upper::Bool)
+
+Returns views to various bits of the lower triangle of `A` according to the
+`level2partition` procedure defined in [1] if `upper` is `false`. If `upper` is `true` then
+the transposed views are returned from the upper triangle of `A`.
+"""
+function level2partition(A::AbstractMatrix, j::Integer, upper::Bool)
+    n = checksquare(A)
+    @boundscheck checkbounds(1:n, j)
+    if upper
+        r = view(A, 1:j-1, j)
+        d = view(A, j, j)
+        B = view(A, 1:j-1, j+1:n)
+        c = view(A, j, j+1:n)
+    else
+        r = view(A, j, 1:j-1)
+        d = view(A, j, j)
+        B = view(A, j+1:n, 1:j-1)
+        c = view(A, j+1:n, j)
+    end
```

Comment:
This mighjt be one of the rare times i suggest wrapping the block in `@views`

Also didn't we just do out bounds check so we can now `@inbounds` it.
(again rare time i am going to suggest doing that to a whole l block)

Reply:
The only problem with `@views` is that AFAICT you aren't guaranteed to actually get a view, since it uses `Base.maybeview` rather than replacing `getindex` with `view`.

Reply:
Isn't that ok though?
Or does this need a `view`?

Reply:
Not sure, I just wanted to make sure this was the same as Nabla's implementation.